tests/integration/conftest.py

```diff
@@ -288,3 +289,68 @@ def factory(path: Path) -> ConfigUtils:
         return ConfigUtils(path)
 
     return factory
+
+
+@pytest.fixture
+def get_system_release(shell):
+    """
+    This function to be used whenever we need live information about system release.
+    E.g. after conversion system release check.
+    Otherwise, use hardcoded SYSTEM_RELEASE envar from /plans/main.fmf
+    This fixture returns a string of ID and VERSION_ID from /etc/os-release.
+    If /etc/os-release is not available, /etc/system-release is read instead.
+    These could be in generally used for OS specific conditioning.
+    Mapping of OS to ID:
+        {
+            "Centos Linux": "centos",\n
+            "Oracle Linux": "oracle",\n
+            "Alma Linux": "almalinux",\n
+            "Rocky Linux": "rocky"
+        }
+    Examples:
+        Centos Linux 7.9 => centos-7.9\n
+        Oracle Linux 8.6 => oracle-8.6\n
+        Alma Linux 8.4 => almalinux-8.4\n
+        Rocky Linux 8.5 => rocky-8.5
+    """
+    path = Path("/etc/system-release")
+
+    if not path.exists():
+        path = Path("/etc/os-release")
+        with open(path) as osrelease:
+            os_release = {}
+            for line in osrelease:
+                if not re.match(line, "\n"):
+                    key, value = line.rstrip().split("=")
+                    os_release[key] = value
+            system_name = os_release.get("ID").strip('"')
+            system_version = os_release.get("VERSION_ID").strip('"')
+
+    else:
+        with open(path) as sysrelease:
+            sysrelease_as_list = sysrelease.readline().rstrip().split(" ")
+            system_name = sysrelease_as_list[0].lower()
+            for i in sysrelease_as_list:
+                if re.match(r"\d", i):
+                    system_version = i
+
+    if system_name == "ol":
+        system_name = "oracle"
+    system_release = f"{system_name}-{system_version}"
+
+    return system_release
+
+
+@pytest.fixture(scope="function")
+def install_required_packages(shell):
+    """
+    Installs packages based on values under TEST_REQUIRES envar in tmt metadata, when called.
+    """
+    try:
+        required_packages = os.environ.get("TEST_REQUIRES").split(" ")
+        for package in required_packages:
+            print(f"\nPREPARE: Installing required {package}")
```

Comment:
f-strings are not available in python 2.7, how that would work when we run the RHEL7 tests, and we would need to call this function?

Reply:
Hmm, good question. We are actually installing python3 as a part of a preparation step on all distros:
https://github.com/oamg/convert2rhel/blob/9cfb40bc571da21e1e1f1c0f60c373519f2803de/tests/ansible_collections/roles/install-testing-deps/tasks/main.yml#L2 and then use the pip3 to install pytest. I presume that when calling pytest we are actually using the python3 version?

Reply:
Good question indeed, however as Martin mentioned, we are using python3 for the execution and it works fine.
It is used in the sub-man-rollback test
https://github.com/oamg/convert2rhel/blob/4b97a984319328221d080e9a54ec7d698216b524/tests/integration/tier0/sub-man-rollback/main.fmf#L20

Reply:
Shouldn't be better to run the code against the same python version that we use for the development/real world? There are a couple of things that can arise on python 2.7 that will not show in python 3, for example.

Reply:
I will take this to our QE sync on Monday

Reply:
I'd say that in the case of integration tests it doesn't matter if the testing framework uses Python 2, Python 3 or Bash. We execute convert2rhel and then observe what it outputs and what convert2rhel does to the system. The integration tests don't need to care much if convert2rhel is written in Python, C++, or Fortran.

@abadger, am I right in the assessment?

Reply:
I agree, @bocekm .  The convert2rhel context manager that we use in the integration tests runs convert2rhel in a separate subprocess.  So that means the convert2rhel code will run under python2 even if pytest is run under python3.  If there were any integration tests that imported convert2rhel code and executed it that way, using the system python would then be a concern.